File: src/com/eagle/programmar/COBOL/COBOL_Variable.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 27, 2013

package com.eagle.programmar.COBOL;

import com.eagle.generate.TypeEnum;
import com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
import com.eagle.programmar.COBOL.Symbols.COBOL_Modifiable_Identifier;
import com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
import com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
import com.eagle.tokens.AbstractToken;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;

public class COBOL_Variable extends TokenChooser
{
	public @CHOICE COBOL_KeywordChoice XXRETURNCODE = new COBOL_KeywordChoice("RETURN-CODE");

	public @CHOICE static class COBOL_UserVariable extends TokenSequence
	{
		public @S(10) COBOL_Modifiable_Identifier id;
		public @S(20) @OPT TokenList<COBOL_Subscript> subscript;
		public @S(30) @OPT TokenList<COBOL_OfVariable> ofList;

		public static class COBOL_OfVariable extends TokenSequence
		{
			public @S(10) COBOL_Keyword OF = new COBOL_Keyword("OF");
			public @S(20) COBOL_Identifier_Reference id;
		}
		
		public TypeEnum findDefinitionType()
		{
			// Find it's declaration
			AbstractToken parent = this.getParent();
			while (parent != null)
			{
				if (parent instanceof COBOL_Program_Complete)
				{
					COBOL_Program_Complete complete = (COBOL_Program_Complete) parent;
					return complete.findDefinitionType(this.id);
				}
				parent = parent.getParent();
			}
			return TypeEnum.OTHER;	// Can't figure out what type it is
		}
	}

	public static String repairName(String cobolVariable)
	{
		return cobolVariable.replaceAll("-", "_");
	}
}

File: csrc/com/eagle/programmar/COBOL/COBOL_Variable.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Nov 27, 2013

namespace com.eagle.programmar.COBOL
{
	using COBOL_Identifier_Reference = com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference;
	using COBOL_Modifiable_Identifier = com.eagle.programmar.COBOL.Symbols.COBOL_Modifiable_Identifier;
	using COBOL_Keyword = com.eagle.programmar.COBOL.Terminals.COBOL_Keyword;
	using COBOL_KeywordChoice = com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenList = com.eagle.tokens.TokenList;
	using TokenSequence = com.eagle.tokens.TokenSequence;

	public class COBOL_Variable : TokenChooser
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE COBOL_KeywordChoice XXRETURNCODE = new com.eagle.programmar.COBOL.Terminals.COBOL_KeywordChoice("RETURN-CODE");
		public COBOL_KeywordChoice XXRETURNCODE = new COBOL_KeywordChoice("RETURN-CODE");

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class COBOL_UserVariable extends com.eagle.tokens.TokenSequence
		public class COBOL_UserVariable : TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Symbols.COBOL_Modifiable_Identifier id;
			public COBOL_Modifiable_Identifier id;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT TokenList<COBOL_Subscript> subscript;
			public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) @OPT TokenList<COBOL_OfVariable> ofList;
			public  OPT;

			public class COBOL_OfVariable : TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.COBOL.Terminals.COBOL_Keyword OF = new com.eagle.programmar.COBOL.Terminals.COBOL_Keyword("OF");
				public COBOL_Keyword OF = new COBOL_Keyword("OF");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.COBOL.Symbols.COBOL_Identifier_Reference id;
				public COBOL_Identifier_Reference id;
			}
		}

		public static string repairName(string cobolVariable)
		{
			return cobolVariable.replaceAll("-", "_");
		}
	}

}
